### ISSUE TITLE
Build errors on FreeBSD 11

### ISSUE THREAD

**Issue by JX7P** [open-watcom/open-watcom-v2#184]
Evening,

I experience some build errors with OpenWatcom on FreeBSD 11-Current.

The first of note is that it cannot reach builder due to being unable to find gcc. When switched to clang mode it instead complains about being unable to find libsupc++ and libstdc++; replacing references to those with -lc++ produces problems as well.

The next is that a file which makes reference to off_t (it #defines something as off_t) appears to fail to include unistd.h, which defines off_t on FreeBSD. This fails at some point in the build process where it is included by a C file that does not include unistd.h before it.

Finally, I am stalled at this:
```**** BUILD rule
============= 19:56:10 /ws/Misc/open-watcom-v2-master/bld/browser =============
=========== 19:56:10 /ws/Misc/open-watcom-v2-master/bld/browser/gen ===========
======= 19:56:10 /ws/Misc/open-watcom-v2-master/bld/browser/brg/dos386 ========
======== 19:56:10 /ws/Misc/open-watcom-v2-master/bld/browser/brg/nt386 ========
======= 19:56:10 /ws/Misc/open-watcom-v2-master/bld/browser/brg/os2386 ========
========== 19:56:10 /ws/Misc/open-watcom-v2-master/bld/browser/nt386 ==========
dosbox -noautoexec -c "mount c /ws/Misc/open-watcom-v2-master/docs" -c "mount d ." -c "mount e /ws/Misc/open-watcom-v2-master/bld" -c "d:wgml.bat" -noconsole
DOSBox version 0.74
Copyright 2002-2010 DOSBox Team, published under GNU GPL.
---
CONFIG:Loading primary settings from config file /home/david/.dosbox/dosbox-0.74.conf
MIDI:Opened device:none
DOSBox switched to max cycles, because of the setting: cycles=auto. If the game runs too fast try a fixed cycles amount in DOSBox's options.
Abort
mv: rename TMP.PTF to wbrw.ptf: No such file or directory
Error(E42): Last command making (compile_gml_to_xxx;wbrw.h) returned a bad status
Error(E02): Make execution terminated
Error(E42): Last command making (wbrw) returned a bad status
Error(E02): Make execution terminated
Error(E42): Last command making (/ws/Misc/open-watcom-v2-master/docs/nt/wbrw.h) returned a bad status
Error(E02): Make execution terminated
<pmake -d build         -h> => non-zero return: 512
Build failed```

This is documentation related, it seems; that I `export OWDOCBUILD=0` in my local setvars.sh copy does not appear to affect this issue, and neither does setting OWDOCBUILD=1 help.

**Comment by jmalak** [comment]
Thanks for your bug report.
I already test gcc support for BSD (9.x) build but not Clang.
I fixed main build make files for better support Clang 65c135822412ea109331bde05251e24a67bee4cf and 0596ea0bade7d513d326de37953991f3b26d9308.
But there is still problem with C++ code to be portable, it will require some afford.

**Comment by IRIXUser** [comment]
Thanks @jmalak, this has fixed some issues with build now. I have since found a few more issues:

* on line 914 of bld/hcwin/cpp/parsing.cpp, Clang complains of a comparison between a pointer and a character

* build halts: ```
===== 15:33:22 /ws/Misc/open-watcom-v2/bld/cpplib/contain/generic.086/mc ======
c++ wcexcept.obj
c++ wchash.obj
Segmentation fault
Error(E42): Last command making (wchash.obj) returned a bad status
Error(E02): Make execution terminated
<pmake -d build         -h> => non-zero return: 512
Build failed
```

**Comment by jmalak** [comment]
I must review a little bit build make files to properly handle unsupported platforms and use native compilers for build OW.
Anyway thanks for your info. I have now installed FreeBSD 11 on VM that I am able to do some testing.
It looks like on FreeBSD is problem with building DLL import libraries caused by wlink (it call wlib tool for this task), but something is wrong because import library is blank (no any entries).

**Comment by stsp** [comment]
I have the same problem on linux.
Same crash, same stack trace.
If you have any problems with reproducing it,
here's the shell script that will reproduce it for
you.

[a.sh.txt](https://github.com/open-watcom/open-watcom-v2/files/253672/a.sh.txt)

**Comment by stsp** [comment]
Here's the fix:
[0001-fix-crash-on-build.patch.txt](https://github.com/open-watcom/open-watcom-v2/files/253686/0001-fix-crash-on-build.patch.txt)

**Comment by stsp** [comment]
Please run valgrind on your code - it seems to
be finding many errors like this.

**Comment by stsp** [comment]
I think this is fixed.
My testing shows it is fixed by commit 4f92b79,
which seems very odd, but it works.

**Comment by jmalak** [comment]
My check on FreeBSD looks OK. I am closing this issue.
